test/VoteDelegate.t.sol

```diff
@@ -0,0 +1,241 @@
+
+        vm.prank(delegate); proxy.lock(100 ether);
+        vm.prank(delegator1); proxy.lock(10_000 ether);
+
+        assertEq(gov.balanceOf(address(chief)), initialMKR + 10_100 ether);
+
+        address[] memory yays = new address[](1);
+        yays[0] = c1;
+        vm.prank(delegate); proxy.vote(yays);
+        assertEq(chief.approvals(c1), 10_100 ether);
+        assertEq(chief.approvals(c2), 0 ether);
+
+        address[] memory _yays = new address[](1);
+        _yays[0] = c2;
+        vm.prank(delegate); proxy.vote(_yays);
+        assertEq(chief.approvals(c1), 0 ether);
+        assertEq(chief.approvals(c2), 10_100 ether);
+    }
+
+    function testDelegatePolling() public {
+        // We can't test much as they are pure events
+        // but at least we can check it doesn't revert
+
+        vm.prank(delegate); proxy.votePoll(1, 1);
+
+        uint256[] memory ids = new uint256[](2);
+        ids[0] = 1;
+        ids[1] = 2;
+        uint256[] memory opts = new uint256[](2);
+        opts[0] = 1;
+        opts[1] = 3;
+        vm.prank(delegate); proxy.votePoll(ids, opts);
+    }
+
+    function testDelegateVotingFuzz(uint256 wad_seed, uint256 wad2_seed) public {
+        uint256 wad = wad_seed < 1 ether ?  wad_seed += 1 ether : wad_seed % 100 ether;
+        uint256 wad2 = wad2_seed < 1 ether ?  wad2_seed += 1 ether : wad2_seed % 20_000 ether;
+        uint256 initialMKR = gov.balanceOf(address(chief));
+
+        vm.prank(delegate); gov.approve(address(proxy), type(uint256).max);
+        vm.prank(delegator2); gov.approve(address(proxy), type(uint256).max);
+
+        uint256 delGovBalance = gov.balanceOf(address(delegate));
+        uint256 del2GovBalance = gov.balanceOf(address(delegator2));
+
+        vm.prank(delegate); proxy.lock(wad);
+        vm.prank(delegator2); proxy.lock(wad2);
+
+        assertEq(gov.balanceOf(address(delegate)), delGovBalance - wad);
+        assertEq(gov.balanceOf(address(delegator2)), del2GovBalance - wad2);
+        assertEq(proxy.stake(address(delegate)), wad);
+        assertEq(proxy.stake(address(delegator2)), wad2);
+        assertEq(gov.balanceOf(address(chief)), initialMKR + wad + wad2);
+
+        address[] memory yays = new address[](1);
+        yays[0] = c1;
+        vm.prank(delegate); proxy.vote(yays);
+        assertEq(chief.approvals(c1), wad + wad2);
+        assertEq(chief.approvals(c2), 0 ether);
+
+        address[] memory _yays = new address[](1);
+        _yays[0] = c2;
+        vm.prank(delegate); proxy.vote(_yays);
+        assertEq(chief.approvals(c1), 0 ether);
+        assertEq(chief.approvals(c2), wad + wad2);
+    }
+
+    function testRevertsDelegateAttemptsSteal() public {
+        vm.prank(delegate); gov.approve(address(proxy), type(uint256).max);
+        vm.prank(delegator1); gov.approve(address(proxy), type(uint256).max);
+
+        vm.prank(delegate); proxy.lock(100 ether);
+        vm.prank(delegator1); proxy.lock(10_000 ether);
```

Comment:
I think we don't need `delegator1` in this test if we are checking the exact revert reason.

```suggestion
        vm.prank(delegate); proxy.lock(100 ether);
```